lib/activerecord/delay_touching/state.rb

```diff
@@ -43,8 +43,13 @@ def more_records?
       end
 
       def add_record(record, *columns)
-        columns << nil if columns.empty? #if no arguments are passed, we will use nil to infer default column
+        # Inferring nil for touch calls with no column specified is creating duplicate DB calls for nested records
+        # Grab the default timestamp columns now as opposed to at write time
+        columns = record.send(:timestamp_attributes_for_update_in_model) if columns.empty?
+
         columns.each do |column|
+          # Convert column explicitly to string here to keep types consistent
+          column = column.to_s
```

Comment:
Probably fine, but in the case that `column` is ever nil, it will be cast to `''`, and that may affect behavior downstream. 

I don't know whether `*columns* can ever contain nil, so FYI.

Depending on what the intention is here (are you trying to resolve symbols and strings?), you could do `column = column.to_s if column` similar to some of the other lines we've seen.

Again, would be good to cover in a spec or something normally, but 🤷‍♂️ 

Probably fine as is in practice, too.

Reply:
I don't think it's possible to get a `nil` value here:
- In ActiveRecord, `timestamp_attributes_for_update_in_model()` will always return a hardcoded array of `[:updated_at, :updated_on]`
- If `nil` is provided as the column, we will replace it out with the attributes that come back from `timestamp_attributes_for_update_in_model()`
- If someone passes in a totally bogus column in the first place, they're going to run into active record issues downstream anyways, and that code should never be released to production

I could add a spec that expects an active record exception to be raised in the event that a bad column value is passed in here, but I'm not totally sold on that being valuable. Let me know what you think!

Reply:
Yeah, fine for now. I suspected that would be the case and mentioned that I didn't think this would happen in practice. Just FYI, and in case I missed anything. 

> If nil is provided as the column, we will replace it out with the attributes that come back from timestamp_attributes_for_update_in_model()
Where does this happen? Looks i

Reply:
In this function on line 48. That's the primary change we're making in this function in this PR: https://github.com/goatapp/activerecord-delay_touching/pull/5/files#diff-c3d5a91eaad03c43d84f142a4473a06dR48

Reply:
FWIW, L48 will take effect if columns == [], but not if columns == [nil], and my concern was over what happens if one of the elements in that array is nil.